project/Settings.scala

```diff
@@ -557,4 +557,21 @@ object Settings {
 
   lazy val javaMajorVer = sys.props("java.version").takeWhile(_.isDigit).toInt
 
+  def classloadersForCustomProtocolTest(fromProject: Project): Setting[Seq[Task[Seq[java.io.File]]]] = 
+    Test / sourceGenerators += Def.task {
+      val customLoaderClasspath = (fromProject / Compile / fullClasspath).value
+      val dq = '"'
+      val files = customLoaderClasspath.files.map(f => 
+        dq + f.toURI().toURL().toString + dq
```

Comment:
The `toURL` is unnecessary here
```suggestion
        dq + f.toURI.toString + dq
```